### PR TITLE
makes /datum/gas_mixture/react faster

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
image

WIP because I forgot something, I distinctly remember doing this approach in an earlier branch and getting a bit more out of it. Also still contains the old procs for testing and all sorts of junk.

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
also i'm going to roll the whole thing into SSair.gas_reactions rather than having the glob meme

**Comment by optimumtact** [comment]
by god he's done it again

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -407,7 +410,69 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
...
+
+			for(var/id in min_reqs)
+				if (id == "TEMP" || id == "ENER")
+					continue
+				if(!cached_gases[id] || cached_gases[id][MOLES] < min_reqs[id])
+					continue reaction_loop
+			//at this point, all minimum requirements for the reaction are satisfied.
+
+			/*	currently no reactions have maximum requirements, so we can leave the checks commented out for a slight performance boost
+				PLEASE DO NOT REMOVE THIS CODE. the commenting is here only for a performance increase.
```
Comment (ninjanomnom) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: `#ifdef`?

Comment (Cyberboss) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm]: not even, it's just literally unused code rn, but may be in the future

Comment (vuonojenmustaturska) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm]: it's unused code dunc wants to keep around

**Comment by vuonojenmustaturska** [comment]
still needs a testmerge but has survived metastation-wide fusion

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 2]
```diff
@@ -410,22 +410,26 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
...
-	var/possible
+	var/topprio // priority of the first element in the list
+	var/list/reactions = list()
 	for(var/I in cached_gases)
-		if(GLOB.nonreactive_gases[I])
-			continue
-		possible = TRUE
-		break
-	if(!possible)
+		for(var/R in SSair.gas_reactions[I])
```
Comment (Cyberboss) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: Does this have to be done everytime?

**Comment by vuonojenmustaturska** [comment]
image

**Comment by vuonojenmustaturska** [comment]
image

oh yeah this is what the reactions list actually looks like now, this probably makes more sense than the code

**Comment by vuonojenmustaturska** [comment]
tested locally and reactions including hypernob still work

**Pull request merged**